### DISCUSSION TITLE
Ignoring specific checks for automerge

### DISCUSSION THREAD

**Discussion by jwodder** [renovatebot/renovate#29147] [Request Help]
### What would you like help with?

I would like help with my configuration

### How are you running Renovate?

Mend Renovate hosted app on github.com

### Please tell us more about your question or problem

Some of my projects have nondeterministic tests that can cause the measured code coverage to vary slightly on separate runs, even on PRs that don't touch the code (such as Renovate's dependency-bumping PRs).  This means that a Renovate PR could end up with one or two failing Codecov checks, yet I do not want such failures to prevent automerging; on the other hand, I do want failures of the actual GitHub Actions tests to prevent automerging.  How can I achieve this result?

Based on my reading of Renovate's documentation, I suspect that the following setup should work:

- Set `automerge` to `true`
- Keep `platformAutomerge` at `true`
- Set up branch protection for the repository's default branch, with "Require status checks to pass before merging" enabled and all GitHub Actions checks (but not the Codecov checks) marked as required.

Unfortunately, when I tried to test this setup out, the coverage checks kept passing over & over, so I was not able to observe what happens when they fail on a Renovate PR.  Rather than risking finding out the hard way that this setup doesn't work several months from now, I thought I'd ask for confirmation from the Renovate experts.

### Logs (if relevant)

<details><summary>Logs</summary>

```

Replace this text with your logs, between the starting and ending triple backticks

```

</details>

**Comment by HonkingGoose** [comment]
> Unfortunately, when I tried to test this setup out, the coverage checks kept passing over & over, so I was not able to observe what happens when they fail on a Renovate PR. Rather than risking finding out the hard way that this setup doesn't work several months from now, I thought I'd ask for confirmation from the Renovate experts.

Your configuration sounds good to me. :wink: 

Quote from [Renovate docs, `platformAutomerge` config option](https://docs.renovatebot.com/configuration-options/#platformautomerge):

> [!NOTE]
> If you use the default `platformAutomerge=true` then you should enable your Git hosting platform's capabilities to enforce test passing before PR merge. If you don't do this, the platform might merge Renovate PRs even if the repository's tests haven't started, are in still in progress, or possibly even when they have failed. On GitHub this is called "Require status checks before merging", which you can find in the "Branch protection rules" section of the settings for your repository. [GitHub docs, about protected branches](https://docs.github.com/en/repositories/configuring-branches-and-merges-in-your-repository/defining-the-mergeability-of-pull-requests/about-protected-branches) [GitHub docs, require status checks before merging](https://docs.github.com/en/repositories/configuring-branches-and-merges-in-your-repository/defining-the-mergeability-of-pull-requests/about-protected-branches#require-status-checks-before-merging) If you're using another platform, search their documentation for a similar feature.

This means you should:

- Set `automerge` to `true`, double check you're putting this in the _correct_ location, so you don't accidentally automerge things you want to manually review first! :wink: 
- Keep `platformAutomerge` to `true`, to let GitHub control when you (or Renovate) are allowed to merge
- Use GitHub's branch protection rules to set which checks are required before you (or Renovate) can merge the PR into the protected branch:
  - Put the critical tests _in_ the required checks section
  - But leave the code coverage test _out_ of the required tests, so they do not block you (or Renovate) from merging

And a _optional_ step: enable GitHub Merge Queue to speed up automerges, read the [Renovate docs, key concepts automerge, GitHub Merge Queue](https://docs.renovatebot.com/key-concepts/automerge/#github-merge-queue) to learn more.
